nix/tools/mkShell.nix

```diff
@@ -0,0 +1,40 @@
+# This Nix expression appends/modifies an existing attribute set
+# in order to run scripts/setup if needed, as well as define STATUS_REACT_HOME
+
+{ pkgs, stdenv ? pkgs.stdenv }:
+
+# Declare a specialized mkShell function which adds some bootstrapping
+#  so that e.g. STATUS_REACT_HOME is automatically available in the shell
+attrs:
+  (pkgs.mkShell.override({ inherit stdenv; }) attrs)
+    .overrideAttrs(super: rec {
+      nativeBuildInputs = (super.nativeBuildInputs or [ ]) ++ [ pkgs.git ];
+
+      # avoid terinal issues
+      TERM="xterm";
+
+      # default locale
+      LANG="en_US.UTF-8";
+      LANGUAGE="en_US.UTF-8";
+
+      shellSetup = ''
+        set -e
+
+        if [ -z "$STATUS_REACT_HOME" ]; then
+          export STATUS_REACT_HOME=$(git rev-parse --show-toplevel)
+        fi
+
+        export SHELL_SETUP=done
```

Comment:
Where do you see it being run?

Reply:
Just add an `echo` here and you'll notice it gets output every single time you call `make shell`: https://github.com/status-im/status-react/blob/nix-refactor/nix/tools/mkShell.nix#L34

I guess currently it's not an issue since `setup` is lightweight, but if it were to become heavy again we'd have to fix this.

Reply:
But `nix/tools/mkShell.nix` is not calling `scripts/setup` at all.

Reply:
Ah I see, we're only calling it from `shell.sh` if the `nix` command isn't available. I guess that's fine then. This command needs updating then: https://github.com/status-im/status-react/blob/dcae15ee0b5b5ffc318d1ea94f84c7796e6f471c/nix/tools/mkShell.nix#L2

Reply:
done